ppl/archive/import.go

```diff
@@ -44,55 +54,125 @@ func Import(ctx context.Context, ark *Archive, zctx *resolver.Context, r zbuf.Re
 // TODO zq#1432 Writer does not currently keep track of size of records
 // written to temporary files. At some point this should have a maxTempFileSize
 // to ensure the Writer does not exceed the size of a provisioned tmpfs.
-//
-// TODO zq#1433 If a tsDir never gets enough data to reach ark.LogSizeThreshold,
-// the data will sit in the tsDirWriter and remain unsearchable until the
-// provided read stream is closed. Add some kind of timeout functionality that
-// periodically flushes stale tsDirWriters.
 type Writer struct {
-	ark     *Archive
-	ctx     context.Context
-	writers map[tsDir]*tsDirWriter
+	ark          *Archive
+	cancel       context.CancelFunc
+	ctx          context.Context
+	errgroup     *errgroup.Group
+	flushTimeout time.Duration
+	once         sync.Once
+	writers      map[tsDir]*tsDirWriter
+	writersMu    sync.RWMutex
 
 	memBuffered int64
 	stats       ImportStats
 }
 
+// NewWriter creates a zbuf.Writer compliant writer for writing data to an
+// archive. If the provided flushTimeout timeout is 0, the flushTimeout will be
+// set to the value of DefaultImportFlushTimeout.
 func NewWriter(ctx context.Context, ark *Archive) *Writer {
```

Comment:
Don't forget to add flushTimeout!

Reply:
You call `SetFlushTimeout` method on Writer.